services/retentionmgr/metadataDep.go

```diff
@@ -312,19 +312,60 @@ func (t *metadataDepImpl) MarkExtentConsumed(destID destinationID, extID extentI
 		common.TagExt: string(extID),
 	})
 
-	log.Debug("MarkExtentConsumed: UpdateExtentStats on metadata")
+	log.Info("MarkExtentConsumed: UpdateExtentStats on metadata")
 
 	resp, err := t.metadata.UpdateExtentStats(ctx, req)
 	if err != nil {
 		log.WithField(common.TagErr, err).Error(`MarkExtentConsumed: UpdateExtentStats failed`)
 		return
 	}
 
-	log.WithField(`resp.status`, resp.GetExtentStats().GetStatus()).Debug(`MarkExtentConsumed done`)
+	log.WithField(`resp.status`, resp.GetExtentStats().GetStatus()).Info(`MarkExtentConsumed done`)
+	return
+}
+
+func (t *metadataDepImpl) GetExtentsForConsumerGroup(destID destinationID, cgID consumerGroupID) (extIDs []extentID, err error) {
+
+	req := metadata.NewReadConsumerGroupExtentsLiteRequest()
+	req.ConsumerGroupUUID = common.StringPtr(string(cgID))
+	req.MaxResults = common.Int32Ptr(defaultPageSize)
+
+	ctx, cancel := thrift.NewContext(30 * time.Second)
+	defer cancel()
+
+	log := t.logger.WithFields(bark.Fields{
+		common.TagDst:  string(destID),
+		common.TagCnsm: string(cgID),
+	})
+
+	for {
+		log.Info("GetExtentsForConsumerGroup: ReadConsumerGroupExtentsLite on metadata")
+
+		res, e := t.metadata.ReadConsumerGroupExtentsLite(ctx, req)
+		if e != nil {
+			err = e
+			log.WithField(common.TagErr, err).Error("GetExtentsForConsumerGroup: ReadConsumerGroupExtentsLite failed")
+			break
```

Comment:
return here?

Reply:
i do a 'break' so it will print the "done" log below ..